### ISSUE TITLE
request: CoC should explicitly forbid use of generated content (LLM, AI, etc.)

### ISSUE THREAD

**Issue by ctcpip** [tkellen/tc39-code-of-conduct-proposal#62]
not my request, but I am not opposed to the idea.  while arguably covered by the existing tenets of the CoC, it has been suggested that the CoC should explicitly prohibit the use of LLMs / AI for contributions and participation in TC39 spaces

**Comment by ljharb** [comment]
I'm not convinced it's actually a conduct violation any more than filing a silly question is, or asking in the wrong place. This is just the societal cost of LLMs being so accessible.

We should definitely indicate in contribution guidelines that it's not tolerated/accepted, and we should definitely discourage it.

**Comment by ctcpip** [comment]
agree, a blanket ban would be overly broad.  I'm not sure we could nail down a qualified prohibition either.  open to ideas

**Comment by bakkot** [comment]
I mostly just want to require up-front disclosure when submitting anything (including comments) written by an LLM.

**Comment by ljharb** [comment]
That seems like a perfectly fine thing to add to contributing guidelines - it can be done org-wide, and then also added to individual repos that have their own file.

**Comment by bakkot** [comment]
This [came up again](https://matrixlogs.bakkot.com/TC39_General/2025-05-21#L2). Seems like there's broad support for forbidding LLM-authored content, not to include translation. Let's do it.

**Comment by ljharb** [comment]
I agree we should do it. I strongly disagree that it's a CoC issue.

**Comment by tabatkins** [comment]
It is a CoC issue, in exactly the same way that sending spam or irrelevant commentary is. To the extent that the CoC captures some expected rules of how we interact with each other, this belongs there.

I'm fine with capturing it in the contributing guidelines, and having CoC point to the contributing guidelines as something else that needs to be followed (in addition to the less technical requirements the CoC places on communication)

**Comment by ljharb** [comment]
There's overlap, sure - I guess I'll wait and see what verbiage people propose for the CoC part vs the contributing part, because I can't see how we could word those to exclude the bad usages and not punish the reasonable ones.

**Comment by bakkot** [comment]
Proposal:
[edit: tweaked wording per below]

### Author your own comments

Any contributions or comments must be your own writing, not the product of LLMs or other tools. Do not prompt an LLM to expand on or explain your idea and then post the output; just provide the idea itself.

Machine translation is permissible, including translation by an LLM, but your use of translation should not introduce any new content.

**Comment by tabatkins** [comment]
`s/should/must/`, and maybe make "comments" more generic - "comments and contributions"?. Otherwise lgtm

**Comment by michaelficarra** [comment]
```diff 
- Do not prompt an LLM to expand on your idea and then post the output; just provide the idea itself.
+ Do not prompt an LLM to expand on your idea or explain your idea for you and then post the output; just provide the idea itself.
```

**Comment by ctcpip** [comment]
I would think an acceptable use policy must forbid any technical contributions where an LLM is involved. Strict machine translation is probably fine.

**Comment by MylesBorins** [comment]
I fail to see why this is necessary.

Spam is spam is spam.

If an LLM is used to spam that feels irrelevant to the fact that it is spam. If an LLM could, for example, help a non English speaker engage in good faith would that not be a net benefit?

Similarly, why should contributions leveraging an LLM be banned? I could train a model entirely on historical spec text and leverage that to help me make decisions about changes to the spec. Why should that be banned?

**Comment by bakkot** [comment]
@ctcpip "any technical contributions where an LLM is involved" is definitely too strong. Many people use some form of LLM-based autocomplete without even thinking about it and I don't think we can reasonably ask them not to. Even less insidiously, many people use them as rubber ducks, and I think it would be pretty absurd to ban contributions that you've talked over with an LLM. The property I want to preserve is that you're meaningfully the author of whatever it is you're submitting.

@MylesBorins It's necessary because we've seen a number of occasions when people have submitted issues or comments which were (it eventually became clear) authored by an LLM, and not a particularly coherent one. (See most recently [here](https://es.discourse.group/t/mixing-fifo-and-lifo-in-javascripts-event-loop/2364).) When people respond they feed that into the LLM and then post the LLM's response. These aren't "spam" in the sense of being self-promotion or off-topic or something like that, and there is nothing in the current CoC which clearly prohibits doing this. I want to prohibit this because reading these threads, much less responding, is a massive waste of my time.

I don't have a problem with you leveraging LLMs to make decisions, if you want to. I do have a problem with you submitting things which an LLM wrote. I've tried to narrowly scope the prohibition to the latter case in [my suggestion above](https://github.com/tc39/code-of-conduct/issues/62#issuecomment-2899419932).

**Comment by michaelficarra** [comment]
@MylesBorins The thing we're tying to ban is foisting the burden of filtering out garbage onto TC39 delegates who have to triage these inputs. If you use an LLM for ideas and then write your own proposal, that's fine. If you feed an idea to an LLM and have *it* write a proposal, then post whatever comes out of it to one of our forums, that is *not* fine.

**Comment by ctcpip** [comment]
@bakkot fair points. I didn't intend to paint with that wide of a brush, but I definitely see why I came across that way.

> The property I want to preserve is that you're meaningfully the author of whatever it is you're submitting.

100% agree

**Comment by ljharb** [comment]
I like the proposed text, but that’s a “contributing guidelines” thing, not a CoC thing.

**Comment by bakkot** [comment]
How is that not a CoC thing? The contributing guidelines are stuff like "new proposals go in matrix or esdiscuss" and "you'll have to sign an IPR to submit spec text" and "here is how we use issues and PRs in this specific repository". The CoC lays out how you are expected to conduct yourself in any TC39 space. This is firmly the latter.

**Comment by ljharb** [comment]
Which aspects of the CoC does it violate?

**Comment by bakkot** [comment]
Currently posting LLM-authored comments doesn't clearly violate the CoC, hence this request to modify the CoC so that it would.

**Comment by ljharb** [comment]
A CoC isn’t for regulating all kinds of behavior just because the word “conduct” is present. I am fully on board with discouraging/disallowing LLM-generated content, to be clear, but I’m involved with moderating and interpreting CoCs in many spaces going back many years, and i still firmly believe this is the wrong venue for this guidance/rule.

**Comment by bakkot** [comment]
Can you say more about why?

**Comment by ljharb** [comment]
The point of a CoC on an open source project for me is to govern behavior that typical contribution guidelines wouldn't cover. "Which tools you use to contribute" is part of contribution guidelines, not human behavior.

**Comment by bakkot** [comment]
The point of the proposed rule is not to restrict the use of any particular tool; I wouldn't want people posting the output of Markov chains either. It's to ensure you author your own comments. That's why it says "Author your own comments", not "don't use LLMs".

**Comment by ljharb** [comment]
I agree with that part of it, but I still think that belongs in contribution guidelines, not a CoC.

**Comment by sffc** [comment]
I think ACM has a well-reasoned policy:

https://www.acm.org/publications/policies/frequently-asked-questions

The document says that ACM permits the use of AI to generate new content subject to these conditions:

> - That these systems do not plagiarize, misrepresent, or falsify content in ACM submissions.
> - That the resulting Work in its totality is an accurate representation of the authors’ underlying work and novel intellectual contributions and is not primarily the result of the tool’s generative capabilities.
> - That the authors accept responsibility for the veracity and correctness of all material in their Work, including any computer-generated material.

These would be things that go in the contribution guidelines. Meanwhile, the CoC would say that delegates must conform to these guidelines and disclose their use of AI-generated content.

**Comment by michaelficarra** [comment]
Oops, I forgot to cross-post here: https://matrixlogs.bakkot.com/TC39_Delegates/2025-06-19#L2

**Comment by sffc** [comment]
The slides scheduled for the 2025 July plenary posit this rhetorical question: "Submitting a comment or post under your name but not written by you is clearly bad conduct, right?"

Sure, I don't think anyone would disagree with that.

But, for centuries, it has been very much acceptable to include content written by someone else, so long as the author discloses it and provides a citation.

The spirit of the ACM policy is that authors can include LLM-generated content so long as they disclose it in a similar way.

---

The ACM policy also explicitly calls out Grammarly and similar LLM-based tools that are designed to improve the readability of content without generating new ideas. I think the policy should permit such tools (without the need for citation) in the same way that it permits machine translation.

**Comment by ctcpip** [comment]
worth considering the [ISO guidance](https://www.iso.org/files/live/sites/isoorg/files/developing_standards/who_develops_standards/docs/use%20of%20AI.pdf) as well.  copy/paste:

## General use of AI tools

- a) Do use AI tools, including generative AI, to inspire creativity and guide initial research that informs
the writing and development of ISO standards and other documents but, as always, check the
sources and underlying materials.
- b) Do use AI tools, if you are a Committee Manager or Secretary, to record a meeting for the purposes
of writing meeting minutes as specified in the ISO/IEC Directives Part 1, 2024, Clause SF.10vi
- c) Do use AI tools for translation of non-ISO content to facilitate understanding.
- d) Do be careful with any free or non-licensedvii versions of AI tools by reading the applicable conditions
for your use. When in doubt, talk to your Technical Programme Manager or Committee Manager
who can liaise with the ISO Central Secretariat as needed.
- e) Do familiarize yourself with AI tools before using them to ensure they work as intended.
- f) Do share your experience with AI tools in terms of what works and what does not work with your
Technical Programme Manager and Committee Manager.
- g) Do ensure that data storage, transfer, processing and any necessary destruction of data that has
already served its purpose, follow regulations, best practice and, where applicable, ISO standards
such as ISO/IEC 42001 on management systems for AI.

## Code of conduct for technical work – spotlight on AI

Abiding by the “ISO Code of Ethics and Conduct” (or “Code of conduct”) is an obligation for participation in
ISO technical committees and their subgroups. The below is written to highlight principles from the Code of
conduct in light of the use of AI tools by ISO technical committees. In the case of any conflict between this
document and the Code of conduct, the Code of conduct shall prevail.

- a) Respect and protect the brand and reputation of ISO
Do use AI ethically, in a way that continues to respect cultural, social, and individual sensitivities,
avoiding the potential for harm, misinformation and disinformation.
- b) Work for the net benefit of the international community and uphold the consensus process
Do use AI in a way that benefits the writing and development of International Standards and other
publications while achieving consensus.
- c) Participate actively
Do not use AI to substitute for your attendance, participation, or any other decision-making activity
you undertake in support of writing and developing ISO standards and other documents.
- d) Respect ISO’s standards development policies
Do ensure you are familiar with, and abide by, the ISO Copyright policy and implementation
guidelines, ISO/IEC Directives and ISO Privacy notice, which are mandatorily applicable to ISO
technical work.
Do not use images or text created by generative AI in any ISO content, either internally or externally.
- e) Respect ISO copyright and protected data
Do not feed free, public, or non-licensed AI tools and systems with ISO copyrighted works, personal
data, confidential information, or other sensitive materials.

## Deployment and monitoring in ISO technical committees

- a) Do work together in your technical committee to ensure experts and officers understand the use of
available AI applications and the guidance from ISO.
- b) Do request support from your Technical Programme Manager or the ISO Central Secretariat on the
use of AI and to understand the strengths and limitations of AI tools.
- c) With the help of the ISO Central Secretariat, do ensure that data, if any, used to train AI tools and
systems, as provided by the ISO Central Secretariat will be screened for biases and corrected.
- d) Do verify results or outputs of generative AI, as provided by the ISO Central Secretariat, including
for any unexpected or inappropriate outputs. This may include integrating adequate feedback loops
to support continuous refinement of generative models, if any.

## Training, support, and compliance

- a) Do promote and seek continuous learning to stay updated on AI advancements, ethical concerns,
and best practices with particular emphasis on ISO AI standards and other documents
- b) Do raise concerns or possible misuses of AI to your Technical Programme Manager

**Comment by ctcpip** [comment]
this part particularly stands out:

> Do not use images or text created by generative AI in any ISO content, either internally or externally.

**Comment by bakkot** [comment]
> But, for centuries, it has been very much acceptable to include content written by someone else, so long as the author discloses it and provides a citation.
> The spirit of the ACM policy is that authors can include LLM-generated content so long as they disclose it in a similar way.

The thing is, in forums like the Discourse or Github issues, I don't want to read AI-generated comments even if they are disclosed. I want those comments not to be made at all. It's a waste of everyone's time. Having a disclaimer at the top is better than not, because it means I can just not read the comment, but if people started submitting any other kind of comment that wasn't worth reading with a disclaimer at the top making it clear it wasn't going to be worth reading I'd consider that bad behavior.

In some contexts I'm OK with LLM-generated _code_ as long as human vouches for having carefully reviewed it, because code has function beyond just conveying ideas. But we don't really deal much in code.

> The ACM policy also explicitly calls out Grammarly and similar LLM-based tools that are designed to improve the readability of content without generating new ideas. I think the policy should permit such tools (without the need for citation) in the same way that it permits machine translation.

I'm opposed to the use of tools designed to "improve readability" because many people take that as license to have LLMs basically write their comments. I don't want people to write two sentences and then have an LLM expand those into paragraphs of purple prose. I would strongly prefer to read the original two sentences.

**Comment by bakkot** [comment]
Reading [this post](https://statmodeling.stat.columbia.edu/2025/07/18/i-am-no-longer-chairing-defenses-or-joining-committees-where-students-use-generative-ai-for-their-writing/) (wherein a professor explains why they are no longer willing to supervise grad students who write with LLMs) helped crystalize something for me:

Much of the point of engaging in forums like the Discourse and Github threads, for me, is to give feedback to community members about their ideas and presentation thereof. I do this because humans can learn from feedback and can improve and make better contributions in the future. LLMs cannot. I don't want to volunteer my time and effort in this way unless I am certain that the feedback I'm giving is going to a human.

**Comment by gesa** [comment]
Ecma frequently (but not exclusively) defers to ISO guidelines on things like this. Especially with TC39, which publishes multiple standards with JTC1 fast track, there should be some deference ISO's stance. I don't think they do a great job of making this clear in the document linked above, but their basic guidance is "if you're going to use LLMs, don't show the output to anyone".

**Comment by ctcpip** [comment]
https://github.com/tc39/how-we-work/blob/main/AI_POLICY.md now exists. We did not have consensus to explicitly designate transgressions as Code of Conduct violations.
